File: RepoCoderStudio/repo_explorer_data/sample_repo/policies/TransferPolicy.java
package policies;

import java.util.Set;

/** Repository-specific cross-border transfer policy. */
public class TransferPolicy {
    /** Calculate the repository's 0-100 cross-border transfer risk score. */
    public double transferRiskScore(
        double amount,
        int customerTenureDays,
        String destinationCountry,
        boolean trustedDevice
    ) {
        Set<String> highRiskCountries = Set.of("IR", "KP", "SY");
        double score = 0.0;
        if (amount >= 250000) {
            score += 40;
        } else if (amount >= 100000) {
            score += 25;
        }
        if (customerTenureDays < 30) {
            score += 20;
        }
        if (highRiskCountries.contains(destinationCountry.trim().toUpperCase())) {
            score += 30;
        }
        if (!trustedDevice) {
            score += 15;
        }
        return Math.min(score, 100.0);
    }

    /** Require an additional authentication challenge at score 50 or above. */
    public boolean requiresStepUpAuth(double riskScore) {
        return riskScore >= 50.0;
    }

    /** Return this repository's daily transfer limit for an account tier. */
    public double dailyTransferLimit(String accountTier) {
        return switch (accountTier.trim().toUpperCase()) {
            case "STANDARD" -> 100000.0;
            case "PREMIUM" -> 500000.0;
            case "PRIVATE" -> 2000000.0;
            default -> 50000.0;
        };
    }
}

File: RepoCoderStudio/repo_explorer_data/sample_repo/policies/transfer_policy.py
"""Transfer-risk policy owned by this repository.

The constants are intentionally business-specific: a generic model cannot
infer them from a function signature, while repository retrieval can ground a
change in the exact policy already used by the application.
"""

from __future__ import annotations


def transfer_risk_score(
    amount: float,
    customer_tenure_days: int,
    destination_country: str,
    trusted_device: bool,
) -> float:
    """Calculate the repository's 0-100 cross-border transfer risk score."""
    high_risk_countries = {"IR", "KP", "SY"}
    score = 0.0
    if amount >= 250_000:
        score += 40
    elif amount >= 100_000:
        score += 25
    if customer_tenure_days < 30:
        score += 20
    if destination_country.strip().upper() in high_risk_countries:
        score += 30
    if not trusted_device:
        score += 15
    return min(score, 100.0)


def requires_step_up_auth(risk_score: float) -> bool:
    """Require an additional authentication challenge at score 50 or above."""
    return risk_score >= 50.0


def daily_transfer_limit(account_tier: str) -> float:
    """Return this repository's daily transfer limit for an account tier."""
    tier_limits = {
        "STANDARD": 100_000.0,
        "PREMIUM": 500_000.0,
        "PRIVATE": 2_000_000.0,
    }
    return tier_limits.get(account_tier.strip().upper(), 50_000.0)
